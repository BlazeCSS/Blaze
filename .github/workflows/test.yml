name: CI

on:
  push:
    branches:
      - master

jobs:
  build:
    name: Install, test and build
    
    runs-on: ubuntu-latest
    
    steps:
      - name: Checkout repo
        uses: actions/checkout@master

      - name: Install dependencies
        run: npm ci

      - name: Test
        run: npm run test

      - name: Build site
        run: npm run build
      
      - name: Switch to site dir
        run: cd ./packages/site

      - name: Deploy site
        uses: w9jds/firebase-action@master
        with:
          args: deploy --only hosting
        env:
          FIREBASE_TOKEN: ${{ secrets.FIREBASE_TOKEN }}
          PROJECT_ID: "default"
