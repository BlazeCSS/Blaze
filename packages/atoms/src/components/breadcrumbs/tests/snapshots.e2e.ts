import { snapIt } from '../../../test';

const component = 'breadcrumbs';

describe(component, () => {
  const snap = snapIt(component);

  describe('renders', () => {
    snap('nothing is rendered with no crumbs', '<blaze-breadcrumbs></blaze-breadcrumbs>');
    snap(
      'one crumb',
      `<blaze-breadcrumbs>
      <blaze-breadcrumb>Home</blaze-breadcrumb>
    </blaze-breadcrumbs>`
    );
    snap(
      'two crumbs with a separator',
      `<blaze-breadcrumbs>
      <blaze-breadcrumb>Home</blaze-breadcrumb>
      <blaze-breadcrumb>Sub-section</blaze-breadcrumb>
    </blaze-breadcrumbs>`
    );
    snap(
      'linked crumb',
      `<blaze-breadcrumbs>
      <blaze-breadcrumb href="home.html">Home</blaze-breadcrumb>
      <blaze-breadcrumb href="another-page.html">Another page</blaze-breadcrumb>
      <blaze-breadcrumb>Sub-section</blaze-breadcrumb>
    </blaze-breadcrumbs>`
    );
  });
});
