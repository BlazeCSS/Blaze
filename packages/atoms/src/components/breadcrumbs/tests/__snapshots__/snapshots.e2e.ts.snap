// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`breadcrumbs renders linked crumb 1`] = `"<blaze-breadcrumbs class=\\"hydrated\\"><!----><nav><ol class=\\"c-breadcrumbs\\"> <blaze-breadcrumb href=\\"home.html\\" class=\\"hydrated\\"><!----><li class=\\"c-breadcrumbs__crumb\\"><a class=\\"c-link\\" href=\\"home.html\\">Home</a></li></blaze-breadcrumb> <blaze-breadcrumb href=\\"another-page.html\\" class=\\"hydrated\\"><!----><li class=\\"c-breadcrumbs__crumb\\"><a class=\\"c-link\\" href=\\"another-page.html\\">Another page</a></li></blaze-breadcrumb> <blaze-breadcrumb class=\\"hydrated\\"><!----><li class=\\"c-breadcrumbs__crumb\\">Sub-section</li></blaze-breadcrumb> </ol></nav></blaze-breadcrumbs>"`;

exports[`breadcrumbs renders nothing is rendered with no crumbs 1`] = `"<blaze-breadcrumbs class=\\"hydrated\\"><!----><nav><ol class=\\"c-breadcrumbs\\"></ol></nav></blaze-breadcrumbs>"`;

exports[`breadcrumbs renders one crumb 1`] = `"<blaze-breadcrumbs class=\\"hydrated\\"><!----><nav><ol class=\\"c-breadcrumbs\\"> <blaze-breadcrumb class=\\"hydrated\\"><!----><li class=\\"c-breadcrumbs__crumb\\">Home</li></blaze-breadcrumb> </ol></nav></blaze-breadcrumbs>"`;

exports[`breadcrumbs renders two crumbs with a separator 1`] = `"<blaze-breadcrumbs class=\\"hydrated\\"><!----><nav><ol class=\\"c-breadcrumbs\\"> <blaze-breadcrumb class=\\"hydrated\\"><!----><li class=\\"c-breadcrumbs__crumb\\">Home</li></blaze-breadcrumb> <blaze-breadcrumb class=\\"hydrated\\"><!----><li class=\\"c-breadcrumbs__crumb\\">Sub-section</li></blaze-breadcrumb> </ol></nav></blaze-breadcrumbs>"`;
